### PR TITLE
[Classification] Enable Saliency Map in MPA Classification

### PR DESCRIPTION
**Pull request by eugene123tw**

### CODE REVIEW

File: mpa/cls/inferrer.py [hunk 1]
```diff
@@ -90,26 +97,21 @@ def _infer(self, cfg):
...
+                  saliency_maps = shook.records if dump_saliency_map else [None] * len(self.dataset)
+
+        assert len(eval_predictions) == len(feature_vectors) == len(saliency_maps), \
+                'Number of elements should be the same, however, number of outputs are ' \
+                f"{len(eval_predictions)}, {len(feature_vectors)}, and {len(saliency_maps)}"
+        outputs = dict(
+            eval_predictions=eval_predictions,
+            feature_vectors=feature_vectors,
+            saliency_maps=saliency_maps
+        )
```
Comment (goodsong81) [inline mpa/cls/inferrer.py:117]: Need to insert `model.eval()`

**Review by goodsong81 — CHANGES_REQUESTED** [review]

File: mpa/cls/inferrer.py (hunk 1 above)
Comment (harimkang) [reply mpa/cls/inferrer.py:117]: model.eval() has been reflected in the latest develop branch, so you can rebase it.

File: mpa/cls/inferrer.py [hunk 2]
```diff
@@ -77,10 +81,14 @@ def _infer(self, cfg):
             logger.info('load checkpoint from ' + cfg.load_from)
             _ = load_checkpoint(model, cfg.load_from, map_location='cpu')
 
+        model.eval()
```
Comment (eugene123tw) [inline mpa/cls/inferrer.py:84]: @goodsong81 `eval` added

**Review by goodsong81 — APPROVED** [review]
LGTM.

File: mpa/modules/models/classifiers/sam_classifier.py [hunk 3]
```diff
@@ -252,7 +255,10 @@ def extract_feat(self, img):
         """Directly extract features from the backbone + neck
            Overriding for OpenVINO export with features
         """
-        x = self.featuremap = self.backbone(img)
+        x = self.backbone(img)
+        if torch.onnx.is_in_onnx_export:
```
Comment (JihwanEom) [inline mpa/modules/models/classifiers/sam_classifier.py]: Is it typo? I think this line should be `torch.onnx.is_in_onnx_export()` , always True with current code.

**Review by JihwanEom — CHANGES_REQUESTED** [review]

File: mpa/modules/models/classifiers/sam_classifier.py (hunk 3 above)
Comment (eugene123tw) [reply mpa/modules/models/classifiers/sam_classifier.py]: yes, I missed it. Fixed.

**Review by JihwanEom — COMMENTED** [review]
Could you rebase and test for this PR? #27 is merged today, please check if there is any conflict due to this (especially infer method in classification task).

**Review by JihwanEom — APPROVED** [review]
LGTM. (I confused with OTE PR : https://github.com/openvinotoolkit/training_extensions/pull/1166)

**Pull request merged**